src/jit/morph.cpp

```diff
@@ -2840,290 +2757,222 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* call)
     // On remorph, we grab it from the arg table.
     unsigned numArgs = 0;
 
-    // TODO: this comment is no longer correct, do an appropriate fix.
-    // Process the late arguments (which were determined by a previous caller).
-    // Do this before resetting fgPtrArgCntCur as fgMorphTree(call->gtCallLateArgs)
-    // may need to refer to it.
-    if (reMorphing)
+    // First we need to count the args
+    if (call->gtCallObjp)
     {
-        // TODO: this comment is no longer correct, do an appropriate fix.
-        // We need to reMorph the gtCallLateArgs early since that is what triggers
-        // the expression folding and we need to have the final folded gtCallLateArgs
-        // available when we call RemorphRegArg so that we correctly update the fgArgInfo
-        // with the folded tree that represents the final optimized argument nodes.
-        //
-        // However if a range-check needs to be generated for any of these late
-        // arguments we also need to "know" what the stack depth will be when we generate
-        // code to branch to the throw range check failure block as that is part of the
-        // GC information contract for that block.
-        //
-        // Since the late arguments are evaluated last we have pushed all of the
-        // other arguments on the stack before we evaluate these late arguments,
-        // so we record the stack depth on the first morph call when reMorphing
-        // was false (via RecordStkLevel) and then retrieve that value here (via RetrieveStkLevel)
-        //
-        if (call->gtCallLateArgs != nullptr)
-        {
-            call->gtCallLateArgs = fgMorphTree(call->gtCallLateArgs)->AsArgList();
-            flagsSummary |= call->gtCallLateArgs->gtFlags;
-        }
-        assert(call->fgArgInfo != nullptr);
-        call->fgArgInfo->RemorphReset();
+        numArgs++;
+    }
+    for (GenTree* args = call->gtCallArgs; (args != nullptr); args = args->gtOp.gtOp2)
```

Comment:
nit: IMO, remove the extra parens

Reply:
It's hard to tell, but that was pre-existing. I agree; I'll remove them.

Reply:
Done